crates/signer/src/manager.rs

```diff
@@ -16,125 +23,183 @@ use crate::error::SignerModuleError;
-        msg: &[u8; 32],
+        object_root: &[u8; 32],
     ) -> Result<BlsSignature, SignerModuleError> {
-        let proxy =
-            self.proxy_signers.get(pubkey).ok_or(SignerModuleError::UnknownProxySigner(*pubkey))?;
-        let signature = proxy.signer.sign(self.chain, msg).await;
+        let bls_proxy = self
+            .proxy_signers
+            .bls_signers
+            .get(pubkey)
+            .ok_or(SignerModuleError::UnknownProxySigner(pubkey.to_vec()))?;
+        let signature = bls_proxy.sign(self.chain, *object_root).await;
+        Ok(signature)
+    }
 
+    pub async fn sign_proxy_ecdsa(
+        &self,
+        pubkey: &EcdsaPublicKey,
+        object_root: &[u8; 32],
+    ) -> Result<EcdsaSignature, SignerModuleError> {
+        let ecdsa_proxy = self
+            .proxy_signers
+            .ecdsa_signers
+            .get(pubkey)
+            .ok_or(SignerModuleError::UnknownProxySigner(pubkey.to_vec()))?;
+        let signature = ecdsa_proxy.sign(self.chain, *object_root).await;
         Ok(signature)
     }
 
     pub fn consensus_pubkeys(&self) -> Vec<BlsPublicKey> {
         self.consensus_signers.keys().cloned().collect()
     }
 
-    pub fn proxy_pubkeys(&self) -> &HashMap<ModuleId, Vec<BlsPublicKey>> {
+    pub fn proxy_pubkeys(&self) -> &HashMap<ModuleId, Vec<GenericPubkey>> {
         &self.proxy_pubkeys
     }
 
-    pub fn delegations(&self) -> Vec<SignedProxyDelegation> {
-        self.proxy_signers.values().map(|s| s.delegation).collect()
-    }
-
     pub fn has_consensus(&self, pubkey: &BlsPublicKey) -> bool {
         self.consensus_signers.contains_key(pubkey)
     }
 
-    pub fn has_proxy(&self, pubkey: &BlsPublicKey) -> bool {
-        self.proxy_signers.contains_key(pubkey)
+    pub fn has_proxy(&self, pubkey: &GenericPubkey) -> bool {
+        match pubkey {
+            GenericPubkey::Bls(bls_pk) => self.proxy_signers.bls_signers.contains_key(bls_pk),
+            GenericPubkey::Ecdsa(ecdsa_pk) => {
+                self.proxy_signers.ecdsa_signers.contains_key(ecdsa_pk)
+            }
+        }
     }
 
     pub fn get_delegation(
         &self,
-        proxy_pubkey: &BlsPublicKey,
+        pubkey: &GenericPubkey,
     ) -> Result<SignedProxyDelegation, SignerModuleError> {
-        let signer = self
-            .proxy_signers
-            .get(proxy_pubkey)
-            .ok_or(SignerModuleError::UnknownProxySigner(*proxy_pubkey))?;
-        Ok(signer.delegation)
+        let delegation: Option<SignedProxyDelegation> = match pubkey {
+            GenericPubkey::Bls(bls_pk) => {
+                self.proxy_signers.bls_signers.get(bls_pk).map(|x| x.delegation).map(Into::into)
+            }
+            GenericPubkey::Ecdsa(ecdsa_pk) => {
+                self.proxy_signers.ecdsa_signers.get(ecdsa_pk).map(|x| x.delegation).map(Into::into)
+            }
+        };
+
+        delegation.ok_or(SignerModuleError::UnknownProxySigner(pubkey.as_ref().to_vec()))
     }
 }
```

Comment:
That's the generic implementation of `get_delegation`. If we deem this unwanted as too complex, then we can either:
1) [the simplest] get rid of this method as it's still unused anywhere
2) go the other way and duplicate: `get_delegation_ecdsa` and `get_delegation_bls`
3) go full circle and make the proxy signers, the signers, etc. also generic (this time with a trait bound to the public key instead of the secret keys), but this is going to infest the signer module with generics overall :D